blockchain/proxy.go

```diff
@@ -539,6 +540,37 @@ func (ep *elrondProxy) GetGenesisNodesPubKeys(ctx context.Context) (*data.Genesi
 	return response.Data.Nodes, nil
 }
 
+// GetGuardianData retrieves guardian data from proxy
+func (ep *elrondProxy) GetGuardianData(ctx context.Context, address erdgoCore.AddressHandler) (*api.GuardianData, error) {
+	err := ep.checkFinalState(ctx, address.AddressAsBech32String())
+	if err != nil {
+		return nil, err
+	}
+	if check.IfNil(address) {
+		return nil, ErrNilAddress
+	}
+	if !address.IsValid() {
+		return nil, ErrInvalidAddress
+	}
```

Comment:
Move this before L545. And add test for nil address.

Reply:
thanks, fixed GetAccount as well